CCAHT-65-create-utility-for-adding-query-params-to-a-url

RudraPatel2003 opened the pull request:
This PR adds utility functions to add and remove query parameters to the current URL.
The function definitions are included in queryParams.ts
queryParams.ts also includes documentation on how you can keep track of the current path in a variable, as the implementation can be tricky due to the details of how Next.js's useRouter() works.

To test everything, go to /queryParamsTest
The file that defines that page will be deleted before merging.

zaviermiller commented on pages/queryParamsTest.tsx:
```diff
@@ -0,0 +1,35 @@
+  // This has to due with how nextjs router is made and timings/refreshing the page
+  // The workaround is to wait for the router object itself to update in a useEffect
+  // Here is how I would access the path in code -- I would just use a state variable
+  const router = useRouter()
```
i am not sure about the feasibility of this, but could we create a custom hook for handling this? something like  `useRouterQuery`? that way we have less boilerplate in each component we want to use the utility function.

RudraPatel2003 replied:
That is a very good idea and something I didn't consider. 90e8301 should address this!

zaviermiller reviewed (commented):
take a look at my comment, let me know what you think. i will approve if you don't think that will work

RudraPatel2003 commented:
I have added the useRouterQuery hook and removed the /queryParamsTest page because with our new repo rules, removing it after I get 2 approvals would make me need the approvals again.
To test the new hook, I would just copy over the test page code from 296ae5d to your local repo to see how the test page works.

zaviermiller commented on utils/queryParams.ts:
```diff
@@ -0,0 +1,56 @@
+ * - addURLQueryParam(router, "hi", "test")
+ * - The value of path is now "/test?hi=test"
+ */
+export function useRouterQuery() {
```
awesome!

andlrutt reviewed (approved):
Great job! You'll have to merge with #35 before/after it's merged in, but after that it's good to go
